README.md

```diff
@@ -4,38 +4,128 @@ Recs plugin that allows you to perform replication for adding and removing compo
 
 ## API
 
-**createPlugin()**
+Before you can start using the library, you'll first need to require the ModuleScript from this project.
 
-Creates the plugin that you feed into Recs.
+The ModuleScript will return a function with the following signature:
 
+`Replication(RemoteEvent: remoteEvent) -> ReplicationInstance`
+
+Once you have a ReplicationInstance, you can start to use the following API:
+
+### plugin.client()
+`plugin.client() -> plugin`
+
+Creates a client plugin that you feed into Recs.
+
+*Note*: This will throw unless called from the Client.
+
+#### Example
+```lua
+Recs.Core.new({
+	Replication.plugin.client(),
+})
+```
+
+### plugin.server()
+`plugin.server() -> plugin`
+
+Creates a server plugin that you feed into Recs.
+
+*Note*: This will throw unless called from the Server.
+
+#### Example
 ```lua
 Recs.Core.new({
-	createPlugin(),
+	Replication.plugin.server(),
 })
 ```
 
-**replicate(callback)**
+### replicate(callback)
+`replicate(callback)`
 
 Any calls to a Recs method in this function will replicate to all other players.
+The following calls are supported with this method:
+- Core:addComponent
+	- *The third argument, props, is not yet properly replicated.*
+- Core:removeComponent
+- Core:addSingleton
 
-This can only be used on the server.
+*Note*: This should only be used from the Server.
 
+#### Example
 ```lua
-replicate(function()
-    self.core:addComponent(entity, "Foo")
+Replication.replicate(function()
+	core:addComponent(entity, "Foo")
 end)
 
-replicate(function()
-    self.core:removeComponent(entity, "Foo")
+Replication.replicate(function()
+	core:removeComponent(entity, "Foo")
 end)
 
-replicate(function()
-    self.core:setComponentProps(color, {
-        value = buttonEntity.Color
-    })
+Replication.replicate(function()
+	core:addSingleton("Bar")
 end)
 ```
 
-**replicatePastChanges(player)**
+### replicatePastChanges
+`replicatePastChanges(player)`
+
+Replicates all of the past changes to this Player. This is used for catching up users when they first join.
+
+*Note*: This should only be used from the Server.
+
+#### Example
+```lua
+game:GetService("Players").PlayerAdded:Connect(Replication.replicatePastChanges)
```

Comment:
I think expanding this out to two lines makes it a bit easier to parse.

```lua
local Players = game:GetService("Players")
Players.PlayerAdded:Connect(Replication.replicatePastChanges)
```

Either that, or just referencing `game.Players` directly would be good for a one-liner.